src/overlay/FlowControlCapacity.cpp

```diff
@@ -220,8 +221,15 @@ FlowControlCapacity::msgBodySize(StellarMessage const& msg,
     if (remoteVersion >= Peer::FIRST_VERSION_UPDATED_FLOW_CONTROL_ACCOUNTING &&
         localVersion >= Peer::FIRST_VERSION_UPDATED_FLOW_CONTROL_ACCOUNTING)
     {
-        return static_cast<uint64_t>(xdr::xdr_size(msg) -
-                                     xdr::xdr_size(msg.type()));
+        if (msg.type() == TRANSACTION)
+        {
+            return stellar::getTransactionEnvelopeSize(msg.transaction());
```

Comment:
The goal of the corresponding issue is to have consistency in how we measure tx envelope sizes at different layers. However, computations are still inconsistent (and remain inconsistent): for the purpose of the Soroban resource accounting we use the size of the 'regular' transaction envelope. At the overlay layer we might still deal with fee bump envelopes, for which we still use the inner envelope size for the resource accounting.
IIUC to remove the inconsistency, the helper should correctly handle the fee bump envelopes and use the inner envelope size only. I'm not 100% confident though whether this is the desired behavior at the overlay layer. That also would need the overlay version bump.
I think we need some input from @marta-lokhova as well.

Reply:
thanks @dmkozh, you're right; since fee bumps don't account for the fee-bump envelope itself for resource calculation, overlay needs to support it separately. Special-casing fee bumps here is an option, but it doesn't seem super safe. As we discussed offline, it's probably safer to just add a big enough buffer to overlay limits to ensure that any unaccounted message parts still fit within overlay limits. In this case, we can just go back to using size of the whole message in overlay.